flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -131,6 +131,9 @@ private NettyPartitionRequestClient connectWithRetries(ConnectionID connectionId
     private NettyPartitionRequestClient connect(ConnectionID connectionId)
             throws RemoteTransportException, InterruptedException {
         try {
+            // The highly important to use `sync` here because it waits for this future until it is
+            // done, and rethrows the cause of the failure if this future failed while `await` only
+            // waits for this future to be completed.(without rethrowing the error).
```

Comment:
```suggestion
            // It's important to use `sync` here because it waits for this future until it is
            // done, and rethrows the cause of the failure if this future failed. `await` only
            // waits for this future to be completed, without throwing the error.
```